src/cli/node/json_rpc_client.rs

```diff
@@ -921,92 +928,112 @@ pub fn data_request_report(
+                reveals.push((*pkh, Some(reveal_radon_types)));
+                reveal_txns.push(reveal_transaction);
             }
-        }
-        // Sort reveal list by pkh
-        reveals.sort_unstable_by_key(|(pkh, _)| *pkh);
-        let reveals = reveals;
-
-        let tally = dr_info
-            .tally
-            .as_ref()
-            .map(|t| RadonTypes::try_from(t.tally.as_slice()))
-            .transpose()?;
-
-        (
-            Some(data_request_state),
-            Some(
-                reveals
-                    .into_iter()
-                    .map(|(pkh, reveal)| {
-                        let honest = match dr_info.tally.as_ref() {
-                            None => format!(""),
-                            Some(tally) => {
-                                if tally.out_of_consensus.contains(&pkh)
-                                    && !tally.error_committers.contains(&pkh)
-                                {
-                                    format!("-{}", dr_output.collateral)
-                                } else {
-                                    let reward = tally
-                                        .outputs
-                                        .iter()
-                                        .find(|vto| vto.pkh == pkh)
-                                        .map(|vto| vto.value)
-                                        .unwrap();
-
-                                    let reward = reward - dr_output.collateral;
-
-                                    // Note: the collateral is not included in the reward
-                                    if reward == 0 {
-                                        "0".to_string()
+            for pkh in dr_info.commits.keys() {
+                if !reveals.iter().any(|(reveal_pkh, _)| reveal_pkh == pkh) {
+                    reveals.push((*pkh, None));
+                }
+            }
+            // Sort reveal list by pkh
+            reveals.sort_unstable_by_key(|(pkh, _)| *pkh);
+            let reveals = reveals;
+
+            let tally = dr_info
+                .tally
+                .as_ref()
+                .map(|t| RadonTypes::try_from(t.tally.as_slice()))
+                .transpose()?;
+
+            let mut local_tally = None;
+
+            if let Some(RadonTypes::RadonError(_radon_error)) = tally.as_ref() {
+                // If the result of the data request is `RadError`, we run the tally stage locally
+                // in order to report a better error message.
+                let non_error_min = f64::from(dr_output.min_consensus_percentage) / 100.0;
+                let commits_count = dr_info.commits.len();
+                let report = run_tally_panic_safe(
+                    &reveal_txns,
+                    &dr_output.data_request.tally,
+                    non_error_min,
+                    commits_count,
+                    &witnet_rad::current_active_wips(),
```

Comment:
This should include all active wips except WIP-0018, because we want to conserve the error message.

Reply:
That's tricky, so I opened issue #2072, let's implement it in a separate PR.